dapps/shop/backend/docs/api.md

```diff
@@ -0,0 +1,112 @@
+# API Endpoints
+
+## Authentication & Store Management
+
+Use the following for the test curl commands below:
+
+    export COOKIE_STORAGE="/tmp/test-cookies.txt"
+    export API_ROOT="http://127.0.0.1:3000"
+    export DEFAULT_OPTIONS=(-vk)
+    export JSON_POST_OPTIONS=(-v -X POST -H 'Content-Type: application/json')
+    export JSON_DELETE_OPTIONS=(-v -X DELETE -H 'Content-Type: application/json')
+    export KEY_AUTH_HEADER=(-H 'Authorization: bearer token123')
+
+### `GET /auth`
+
+Check if user is authenticated
+
+    curl "${DEFAULT_OPTIONS[@]}" -b $COOKIE_STORAGE $API_ROOT/auth
+
+### `GET /auth/:email`
+
+Check for an existing seller account by E-mail address
+
+    curl "${DEFAULT_OPTIONS[@]}" $KEY_AUTH_HEADER $API_ROOT/auth/me@example.com
+
+### `POST /auth/login`
+
+Authenticate a seller. Sessions tracked by cookies.
+
+    curl "${JSON_POST_OPTIONS[@]}" -d '{ "email": "test@example.com", "password": "thisismypassword" }' -c $COOKIE_STORAGE $API_ROOT/auth/login
+
+### `POST /auth/registration`
+
+Register a seller account.
+
+    curl "${JSON_POST_OPTIONS[@]}" -d '{ "name": "Test User", "email": "test@example.com", "password": "thisismypassword" }' $API_ROOT/auth/registration
+
+### `DELETE /auth/registration`
+
+Delete a seller account.
+
+    curl -v -X DELETE -b $COOKIE_STORAGE $API_ROOT/auth/registration
+
+### `GET /shop`
+
+List the shops for a seller.
+
+    curl "${DEFAULT_OPTIONS[@]}" -b $COOKIE_STORAGE $API_ROOT/shop
+
+### `POST /shop`
+
+Create a shop with provided configuration.
+
+    curl "${JSON_POST_OPTIONS[@]}" -b $COOKIE_STORAGE -d '{"name": "Test Shop", "listing_id": "1-001-0001", "auth_token": "token123"}' $API_ROOT/shop
+
+### `DELETE /shop`
+
+Delete a shop.
+
+    curl "${JSON_DELETE_OPTIONS[@]}" -b $COOKIE_STORAGE -d '{"id": 123}' $API_ROOT/shop
+
+### `GET /config/dump/:id`
+
+Dump a config.  **NOTE:** Local testing only.
+
+    curl "${DEFAULT_OPTIONS[@]}" -b $COOKIE_STORAGE $API_ROOT/config/dump/1
+
+### `POST /config`
+
+Update a shop config.
+
+    curl $JSON_POST_OPTIONS -b $COOKIE_STORAGE -d '{ "shopId": 1, "config": { "networkId": 4, "stripe_backend": "sk_test_MySecretKeyIsCool", "data_url": "http://ipfs.originprotocol.com/ipfs/Qmc8Esc5eTGi7BP7V6DNa8rDPnRfGh642FAfbvrKkqCVNp/" } }' $API_ROOT/config
```

Comment:
We should convert these casings to be consistent unless there is some reason not to, e.g. stripe_backend -> stripeBackend, data_url -> dataUrl

Reply:
Noticed earlier that sequelize was converting some of their standard cols but not others.  e.g. `createdAt` was in the model but `created_at` in the DB.  Could put together object translators (e.g. model -> output object) or something like that?  Has this been an issue for you?

Reply:
Sequelize should convert all the columns automatically because of the `underscored: true` setting in the config. This is stored as JSON so it won't get converted.

Reply:
Sorry, wasn't paying attention I guess and didn't realize what you were commenting on.  Yeah, this is confusing(it also doesn't represent the full config).  Everything in the config should be underscored and lowercased.  `shopId` is part of the request.  I'm really not sure where `networkId` came from and should probably not be part of the example.  

This was a result of converting from env vars directly to this config.  It made the conversion of dotenv files to JSON config easier than translating underscored to camelCase([ref](https://github.com/OriginProtocol/origin/blob/mikeshultz/dshop-multitenant/dapps/shop/backend/utils/encryptedConfig.js#L210-L223).

I'm open to this being considered a bad call and moving to camelCase.

Reply:
Cleaned up the config JSON example in the docs.  I think that's the minimum fields required for a shop to work as of right now.

Still uses underscore naming though.  You think we should address the naming scheme now?  The change would take a long minute while but much shorter now than dealing with it later.  Might just rule out dotenv file imports without some translation.

Reply:
Yea that'd be great if you have time. Fair bit of mental overhead remembering which ones have to be cased different when writing the client.  You could use a library like `camelcase` for importing dotenv files.

Reply:
Changed the config case stuff.  It'll be a while before I get the public rinkeby backend updated.